onyo/lib/onyo.py

```diff
@@ -38,47 +38,104 @@ def __init__(self, path: Union[Path, str], init: bool = False) -> None:
 
     @property
     def assets(self) -> set[Path]:  # pyre-ignore[11]
+        """
+        A `set` containing the `Path`s to all assets of an Onyo repository.
+
+        If the property is not set, load the assets of the repository and save
+        them in `self._assets`.
+
+        When a function of `Repo` changes assets, the value will be reset to
```

Comment:
This misses the point IMO.

First: "the value will be reset to `None`" is incorrect. From the user's PoV, it is /never/ None.

The /real/ point here is that the cache is guaranteed to remain consistent when `Repo`'s public APIs are used (`set()`, `add()`, etc). Use of private APIs (such as `_git()`) do not guarantee cache correctness. See also: `Repo.clear_caches()`.